m.neural_network.preparetraining/m.neural_network.preparetraining.py

```diff
@@ -0,0 +1,278 @@
+#!/usr/bin/env python3
+
+############################################################################
+#
+# MODULE:       m.neural_network.preparetraining
+#
+# AUTHOR(S):    Guido Riembauer
+#
+# PURPOSE:      TODO.
+#
+# COPYRIGHT:	(C) 2024 by mundialis and the GRASS Development Team
+#
+# 		This program is free software under the GNU General Public
+# 		License (v3). Read the file COPYING that comes with GRASS
+# 		for details.
+#
+#############################################################################
+
+# %Module
+# % description: TODO.
+# % keyword: raster
+# % keyword: vector
+# % keyword: import
+# % keyword: export
+# % keyword: training
+# % keyword: neural network
+# % keyword: preparation
+# %end
+
+# %option G_OPT_M_DIR
+# % key: input_traindir
+# % label: Name of the input training data directory containing subfolders with imagery and labels per tile
+# % guisection: Input
+# %end
+
+# %option G_OPT_M_DIR
+# % key: input_applydir
+# % label: Name of the input apply data directory containing subfolders with imagery per tile
+# % guisection: Input
+# %end
+
+# %option
+# % key: val_percentage
+# % type: integer
+# % required: yes
+# % label: Percentage of training tiles to be used as validation during training
+# % answer: 20
+# % guisection: Input
+# %end
+
+# %option
+# % key: class_column
+# % type: string
+# % required: yes
+# % multiple: no
+# % answer: class_number
+# % label: column of the label vector that holds the class number
+# % guisection: Parameters
+# %end
+
+# %option
+# % key: class_value
+# % type: integer
+# % required: yes
+# % multiple: no
+# % answer: 2
+# % label: Expected and output value for the class of interest
+# % guisection: Parameters
+# %end
+
+# %option
+# % key: no_class_value
+# % type: integer
+# % required: yes
+# % multiple: no
+# % answer: 1
+# % label: Expected and output value for the non class of interest areas (inverse of <class_value>)
+# % guisection: Parameters
+# %end
+
+# %option G_OPT_M_NPROCS
+# %end
+
+import atexit
+from multiprocessing import Pool
+import os
+import random
+
+import grass.script as grass
+from grass_gis_helpers.cleanup import general_cleanup
+from grass_gis_helpers.general import set_nprocs
+from grass_gis_helpers.mapset import verify_mapsets
+from grass.pygrass.modules import Module, ParallelModuleQueue
+from osgeo import gdal
+
+
+# initialize global vars
+ID = grass.tempname(8)
+orig_region = None
+rm_dirs = []
+
+def cleanup():
+    general_cleanup(orig_region=orig_region, rm_dirs=rm_dirs)
+
+def check_parallel_errors(queue):
```

Comment:
Ist das das gleiche wie im preparedata? Wäre es sinnvoll das in eine lib für das addon oder in die grass-gis-helpers aus zu lagern?